tasks/configure-gpg.yml

```diff
@@ -42,44 +41,30 @@
       ignore_errors: "yes"
 
     - name: "Create GPG key when doesn't exist"
-      shell: >
-        {{ archivematica_src_configure_gpg.binary }}
-        --batch
-        --no-tty
-        --homedir={{ archivematica_src_configure_gpg.gpg_home_directory }}
-        --quiet
-        --gen-key
-        /tmp/gpg-batch-file
+      command: "{{ archivematica_src_ss_virtualenv }}/bin/python"
       args:
-        executable: "/bin/bash"
-      become: "yes"
-      become_user: "archivematica"
-      when: gpg_key_already_exist.rc != 0
-      ignore_errors: "yes"
-
-    - name: "Delete GPG tmp file"
-      file:
-        state: absent
-        path: "/tmp/gpg-batch-file"
-      become: "yes"
+        stdin: |
+          import os
+          import gnupg
 
-    - name: "Get gpg key fingerprint"
-      shell: >
-        {{ archivematica_src_configure_gpg.binary }}
-        --no-tty
-        --homedir={{ archivematica_src_configure_gpg.gpg_home_directory }}
-        --quiet
-        --list-secret-keys
-        --fingerprint {{ archivematica_src_configure_gpg.key_user }}
-        | grep "Key fingerprint"
-        | awk -F= '{ print $2}'
-        | tail -n1
-        | sed 's/\s//g'
-      args:
-        executable: "/bin/bash"
+          gpg = gnupg.GPG(
+                  gnupghome = '{{ archivematica_src_configure_gpg.gpg_home_directory }}',
+                  gpgbinary = '{{ gpg_binary }}'
+                )
+          input_data = gpg.gen_key_input(
+            name_email = '{{ archivematica_src_configure_gpg.key_mail }}',
+            name_comment = '{{ archivematica_src_configure_gpg.key_comment }}',
+            name_real = '{{ archivematica_src_configure_gpg.key_user }}',
+            passphrase = '{{ archivematica_src_configure_gpg.key_passphrase }}',
+            key_length = '4096',
+            key_type = 'RSA'
+          )
+          key = gpg.gen_key(input_data)
+          print key
```

Comment:
Does this work `print(key)`? It might need to change with Python 3 otherwise.

Reply:
Yes, it prints the fingerprint and we can take it later with: {{ gpg_fingerprint.stdout }}

This is the -vvvvv task output:

``` 
changed: [am111centos] => {
    "changed": true,
    "cmd": [
        "/usr/share/archivematica/virtualenvs/archivematica-storage-service/bin/python"
    ],
    "delta": "0:00:01.033960",
    "end": "2020-01-29 12:37:13.842298",
    "invocation": {
        "module_args": {
            "_raw_params": "/usr/share/archivematica/virtualenvs/archivematica-storage-service/bin/python",
            "_uses_shell": false,
            "argv": null,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": "import os\nimport gnupg\n\ngpg = gnupg.GPG(gnupghome='/var/archivematica/storage_service')\ninput_data = gpg.gen_key_input(\n  name_email = 'monitor+am111centos@artefactual.com',\n  name_comment = 'Ansible Generated',\n  name_real = 'test',\n  passphrase = '',\n  key_length = '4096',\n  key_type = 'RSA')\nkey = gpg.gen_key(input_data)\nprint key\n",
            "stdin_add_newline": true,
            "strip_empty_ends": true,
            "warn": true
        }
    },
    "rc": 0,
    "start": "2020-01-29 12:37:12.808338",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "2C3CA318A954AC32A0E589BECB31C5004C4394ED",
    "stdout_lines": [
        "2C3CA318A954AC32A0E589BECB31C5004C4394ED"
    ]
}
```

Reply:
@ross-spencer , I just tested with "print(key)" and it works fine. So I'm going to change it.